File: docs/audits/carousel-theme-audit-2026-08-13/proposed-types.swift
import Foundation

// Proposed only: keep dimensions independent so one carousel can participate
// in several useful filters without inventing a single catch-all category.
enum CarouselThemeID: String, Codable, CaseIterable, Sendable {
    case schoolAndNextChapter = "school-and-next-chapter"
    case romanticPartner = "romantic-partner"
    case relationshipsAndMemory = "relationships-and-memory"
    case friendBirthday = "friend-birthday"
    case appreciationAtWork = "appreciation-at-work"
    case homeAndHosting = "home-and-hosting"
    case hobbiesAndPassions = "hobbies-and-passions"
    case outdoorsAndActive = "outdoors-and-active"
    case petPeople = "pet-people"
    case budgetAndValues = "budget-and-values"
    case giftsForHim = "gifts-for-him"
    case giftsForHer = "gifts-for-her"
    case kidsAndTweens = "kids-and-tweens"
    case giftBaskets = "gift-baskets"
    case betterGifting = "better-gifting"
}

enum CarouselOccasion: String, Codable, Sendable {
    case any, anniversary, backToSchool = "back-to-school", birthday
    case carePackage = "care-package", collegeSendoff = "college-sendoff"
    case friendiversary, graduation, holiday, housewarming, justBecause = "just-because"
    case memorial, newPet = "new-pet", nursesWeek = "nurses-week"
    case raceDay = "race-day", teacherAppreciation = "teacher-appreciation"
    case thankYou = "thank-you", valentines, wedding, hostGift = "host-gift"
}

enum CarouselBudget: String, Codable, Sendable {
    case any, budget, mixed, splurge
    case under25 = "under-25", under50 = "under-50"
}

enum CarouselIntent: String, Codable, Sendable {
    case shop, make, learn, inspire
}

enum ThemeConfidence: String, Codable, Sendable {
    case high, medium, low
}

struct CarouselThemeAssignment: Codable, Identifiable, Sendable {
    let id: String
    let filterLabel: String
    let themeId: CarouselThemeID
    let recipients: [String]
    let occasions: [CarouselOccasion]
    let interests: [String]
    let budget: CarouselBudget
    let intent: CarouselIntent
    let confidence: ThemeConfidence
    let navigationExcluded: Bool?
    let evidence: [String]
}

File: docs/audits/carousel-theme-audit-2026-08-13/proposed-types.ts
export const carouselThemeIds = [
  "school-and-next-chapter", "romantic-partner", "relationships-and-memory",
  "friend-birthday", "appreciation-at-work", "home-and-hosting",
  "hobbies-and-passions", "outdoors-and-active", "pet-people",
  "budget-and-values", "gifts-for-him", "gifts-for-her", "kids-and-tweens",
  "gift-baskets", "better-gifting",
] as const;

export type CarouselThemeId = typeof carouselThemeIds[number];
export type CarouselBudget = "any" | "budget" | "mixed" | "splurge" | "under-25" | "under-50";
export type CarouselIntent = "shop" | "make" | "learn" | "inspire";
export type ThemeConfidence = "high" | "medium" | "low";

export type CarouselOccasion =
  | "any" | "anniversary" | "back-to-school" | "birthday" | "care-package"
  | "college-sendoff" | "friendiversary" | "graduation" | "holiday"
  | "host-gift" | "housewarming" | "just-because" | "memorial" | "new-pet"
  | "nurses-week" | "race-day" | "teacher-appreciation" | "thank-you"
  | "valentines" | "wedding";

export interface CarouselThemeAssignment {
  id: string;
  filterLabel: string;
  themeId: CarouselThemeId;
  recipients: readonly string[];
  occasions: readonly CarouselOccasion[];
  interests: readonly string[];
  budget: CarouselBudget;
  intent: CarouselIntent;
  confidence: ThemeConfidence;
  navigationExcluded?: boolean;
  evidence: readonly string[];
}

// CI/runtime boundary: validate JSON with a schema library before casting.
export interface CarouselThemeMap {
  schemaVersion: "1.0.0";
  catalogVersion: string;
  themes: readonly { id: CarouselThemeId; title: string }[];
  carousels: readonly CarouselThemeAssignment[];
}
